skbio/io/_base.py

```diff
@@ -185,3 +186,25 @@ def _format_fasta_like_records(generator, id_whitespace_replacement,
                 "scores associated with it." % cardinal_to_ordinal(idx + 1))
 
         yield header, seq.sequence, seq.quality
+
+
+def _line_generator(fh, skip_blanks=False):
+    for line in fh:
+        line = line.strip()
+        if line or not skip_blanks:
+            yield line
+
+
+def _too_many_blanks(fh, max_blanks):
+    count = 0
+    too_many = False
+    for line in _line_generator(fh):
+        if line:
+            break
+        else:
+            count += 1
+            if count > max_blanks:
+                too_many = True
+                break
+    fh.seek(0)
```

Comment:
Good catch! We forgot about the filehandle copy that happens before sniffing. Let's leave the implementation as-is since `seek`/`tell` is a larger issue that can be dealt with later on. I totally agree with you that it's odd to sniff from the beginning of the file and read from some offset.

I'll take a full pass through the PR later today or tomorrow morning.